server/src/main/java/com/kritsit/casetracker/server/datalayer/UserRepository.java

```diff
@@ -0,0 +1,62 @@
+package com.kritsit.casetracker.server.datalayer;
+
+import java.util.Map;
+
+import com.kritsit.casetracker.server.domain.model.Permission;
+import com.kritsit.casetracker.server.domain.model.Staff;
+
+public class UserRepository implements IUserRepository {
+	
+	private final IPersistenceService db;
+	
+	public UserRepository(){
+		//Just until we have mockin \ DI
+		this.db = new DatabasePersistence();
+	}
+	
+	public UserRepository(IPersistenceService db){
+		this.db = db;
+	}
+	
+    public long getPasswordSaltedHash(String username) throws RowToModelParseException {
+    	try {
+	        String sql = "SELECT passwordHash FROM staff WHERE username=\'" 
+	            + username + "\';";
+	        Map<String, String> rs = db.executeQuery(sql);
+	        return Long.parseLong(rs.get("passwordHash"));
```

Comment:
This will throw an exception if the user enters a username that is not in the database as it will try to call get() from null. This is why the tests were failing